src/libs/actions/PaymentMethods.js

```diff
@@ -114,10 +115,68 @@ function clearDebitCardFormErrorAndSubmit() {
     });
 }
 
+/**
+ * Call the API to transfer wallet balance.
+ * @param {Object} paymentMethod
+ * @param {String|Number} paymentMethod.methodID
+ * @param {'bankAccount'|'debitCard'} paymentMethod.type
+ */
+function transferWalletBalance(paymentMethod) {
+    const parameters = {
+        [paymentMethod.type === CONST.PAYMENT_METHODS.BANK_ACCOUNT ? 'bankAccountID' : 'fundID']: paymentMethod.methodID,
+    };
+    Onyx.merge(ONYXKEYS.WALLET_TRANSFER, {loading: true});
+
+    API.TransferWalletBalance(parameters)
+        .then((response) => {
+            if (response.jsonCode !== 200) {
+                throw new Error(response.message);
+            }
+            Onyx.merge(ONYXKEYS.USER_WALLET, {balance: 0});
+            Onyx.merge(ONYXKEYS.WALLET_TRANSFER, {shouldShowConfirmModal: true, loading: false});
+            Navigation.navigate(ROUTES.SETTINGS_PAYMENTS);
+        }).catch((error) => {
+            Log.alert(`[Payments] Failed to transfer wallet balance: ${error.message}`);
+            Growl.error(Localize.translateLocal('transferAmountPage.failedTransfer'));
+            Onyx.merge(ONYXKEYS.WALLET_TRANSFER, {loading: false});
+        });
+}
+
+/**
+ * Set the transfer account and reset the transfer data for Wallet balance transfer
+ * @param {String} selectedAccountID
+ */
+function saveWalletTransferAccountAndResetData(selectedAccountID) {
+    Onyx.merge(ONYXKEYS.WALLET_TRANSFER, {
+        selectedAccountID,
+        filterPaymentMethodType: null,
+        loading: false,
+        shouldShowConfirmModal: false,
+    });
+}
+
+/**
+ * @param {Number} transferAmount
+ */
+function saveWalletTransferTransferredAmount(transferAmount) {
+    Onyx.merge(ONYXKEYS.WALLET_TRANSFER, {transferAmount});
+}
```

Comment:
I had to store this transferAmount as after user click transfer amount, balance in the userWallet will be set to 0 and then in the ConfirmModal we will not have any way to show the transferred amount.

Reply:
I think `saveWalletTransferAmount` is fine descriptiveness-wise, and not as long :smile:

Reply:
:eyes:  <img width="25" src="https://user-images.githubusercontent.com/24370807/149179470-85df8cb6-b1f0-4776-954c-a3234b5fed00.png" />

:laughing: